index.js

```diff
@@ -86,19 +86,38 @@ export function checkDataIntegrityProofFormat({
               'string', 'Expected "proof.type" to be a string.');
           }
         });
-        it(`"proof.type" field MUST be "${expectedProofTypes.join(',')}".`,
-          function() {
-            this.test.cell = {columnId: vendorName, rowId: this.test.title};
-            for(const proof of proofs) {
-              proof.should.have.property('type');
-              proof.type.should.be.a(
-                'string',
-                'Expected "proof.type" to be a string.'
-              );
-              const hasExpectedType = expectedProofTypes.includes(proof.type);
-              hasExpectedType.should.equal(true);
+        it(`"proof.type" field MUST be "${expectedProofTypes.join(',')}" ` +
+          `and the associated document MUST include expected contexts.`,
+        function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          for(const proof of proofs) {
+            proof.should.have.property('type');
+            proof.type.should.be.a(
+              'string',
+              'Expected "proof.type" to be a string.'
+            );
+            const hasExpectedType = expectedProofTypes.includes(proof.type);
+            hasExpectedType.should.equal(true);
+
+            if(proof.type === 'DataIntegrityProof') {
+              const expectedContexts = [
+                'https://www.w3.org/ns/credentials/v2',
+                'https://w3id.org/security/data-integrity/v2'
+              ];
+              const hasExpectedContexts = expectedContexts.some(
+                value => data['@context'].includes(value));
+              hasExpectedContexts.should.equal(true);
             }
-          });
+
+            if(proof.type === 'Ed25519Signature2020') {
+              const expectedContext =
+                'https://w3id.org/security/suites/ed25519-2020/v1';
+              const hasExpectedContext =
+                data['@context'].includes(expectedContext);
+              hasExpectedContext.should.equal(true);
+            }
+          }
```

Comment:
Seems like we need to have a failure case if the type isn't either of those values. If that's covered elsewhere, ignore this, but if it needs to be here, we'll have to add it.

Reply:
The proof type is being checked here https://github.com/w3c-ccg/data-integrity-test-suite-assertion/blob/3d17ba0f062318ab3bd74b1b69fa4607d29e098e/index.js#L99-L100. I have updated the context checks to run only if the type check passes [3d17ba0](https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/22/commits/3d17ba0f062318ab3bd74b1b69fa4607d29e098e)

Reply:
Oh... if we have that `hasExpectedType.should.equal(true)` assertion then we shouldn't need that extra check in the conditionals. Sorry if I missed that before. Thanks!